### PR TITLE
Use kitipy Context as a ProxyExecutor

### PR DESCRIPTION
**Pull request by NiR-**
A new Executor ABC named BaseExecutor and a ProxyExecutor which is a
basic executor decorator have been introduced. The kitipy Context now
inherits from the ProxyExecutor, such that the Context can be used as if
it was a real executor.

### CODE REVIEW

File: kitipy/context.py [hunk 1]
```diff
@@ -68,9 +68,13 @@ def __init__(self,
         self.config = config
         self.stage = stage
         self.stack = stack
-        self.executor = executor
+        self._executor = executor
```
Comment (neghmurken) [inline kitipy/context.py]: Don't you have to call `super()` here? The `_executor` is already setted in `ProxyExecutor:__init__()`

File: kitipy/context.py [hunk 2]
```diff
@@ -92,87 +96,7 @@ def with_stack(self, stack_name: str):
...
-    def is_remote(self):
-        """Check if current kitipy Executor is in remote mode"""
-        return self.executor.is_remote
-
-    @property
-    def meta(self):
-        """Meta properties from current click.Context"""
-        return click.get_current_context().meta
-
+    # @TODO: fix that
```
Comment (nm2107) [inline kitipy/context.py]: possible to let a more descriptive message ?

File: kitipy/context.py (hunk 1 above)
Comment (NiR-) [reply kitipy/context.py]: Yeah indeed, it's a remainder of a previous implementation.

File: kitipy/context.py (hunk 2 above)
Comment (NiR-) [reply kitipy/context.py]: Haha, yeah I'd like... but I've no idea what I meant :joy:

**Comment by NiR-** [comment]
> Do you intend to introduce more proxy based executors ?

Yep, that's the original idea behind the `ProxyExecutor`. One such use case would be running/execing in a container with something like: `mysqdump(..., _exec=kctx.stack.run_as_exec('db'))` :wink:

**Pull request merged**